syntax = "proto3";

package leapauto.planning;

import "proto/dp_poly_path_config.proto";
import "proto/navi_path_decider_config.proto";

enum TaskType {
  DP_POLY_PATH_OPTIMIZER = 0;
  DP_ST_SPEED_OPTIMIZER = 1;
  QP_SPLINE_PATH_OPTIMIZER = 2;
  QP_SPLINE_ST_SPEED_OPTIMIZER = 3;
  PATH_DECIDER = 4;
  SPEED_DECIDER = 5;
  POLY_ST_SPEED_OPTIMIZER = 6;
  NAVI_PATH_DECIDER = 7;
  NAVI_SPEED_DECIDER = 8;
  NAVI_OBSTACLE_DECIDER = 9;
};

message EMPlannerConfig {
  repeated TaskType task = 1;
  planning.DpPolyPathConfig dp_poly_path_config = 2;
}

message NaviPlannerConfig {
  repeated TaskType task = 1;
  planning.NaviPathDeciderConfig  navi_path_decider_config = 2;
}

message PlanningConfig{

  enum PlannerType {
    RTK = 0;
    EM = 1;  // expectation maximization
    LATTICE = 2;
    NAVI = 3; // navigation planner
  };

  PlannerType planner_type = 1;
  EMPlannerConfig em_planner_config = 2;
  NaviPlannerConfig navi_planner_config = 3;

}
